### DISCUSSION TITLE
Customizing ScalarBar in the client's side

### DISCUSSION THREAD

**Discussion by loisvazquezind** [Kitware/trame#157] [Q&A]
Hello.

I have a problem similar to the one in the discussion [#76](https://github.com/Kitware/trame/discussions/76): I need to edit the color bar, but in my case, I need to do it on the client's side.

As you can see in the image below, the color bars on the server's and client's sides are different. My goal is that the client's bar is the same as the server's. At this point, I don't care about the position itself, but mostly I care about the number of labels, the format of the labels, and the color of the text.

![image](https://user-images.githubusercontent.com/101557949/203559375-b20096df-c254-4708-b261-6165ed236e82.png)

I'm taking [this example](https://github.com/Kitware/trame/blob/master/examples/06_vtk/Applications/MultiFilter/app.py#L162) as code base, to which I added the following code:
```
# Scalar bar
scalarBarActor = vtkScalarBarActor()
scalarBarActor.SetLookupTable(mesh_lut)
scalarBarActor.SetNumberOfLabels(10)
scalarBarActor.SetLabelFormat("%.2E")
scalarBarActor.DrawColorBarOn()
scalarBarActor.SetTitle(default_array.get("text"))
renderer.AddActor2D(scalarBarActor)
```

I have already reviewed the discussions [#101](https://github.com/Kitware/trame/issues/101) and [#5](https://github.com/Kitware/trame-vtk/pull/5), but I think they don't solve my problem (or at least that's how I understood it).

Is it currently possible to do what I describe above?

Thank you very much in advance for the help.

Greetings.

**Comment by jourdain** [comment]
The positioning will indeed be tricky due to the different logic between VTK/C++ and VTK/js. But should be possible is enough time is spend figuring out a mapping. 

For the font color, I'm not sure why they get set to black. But that color is [coming from the C++](https://github.com/Kitware/trame-vtk/blob/master/trame_vtk/modules/vtk/addon_serializer.py#L215).
